pkg/ingester/ingester.go

```diff
@@ -777,10 +789,13 @@ func (i *Ingester) updateActiveSeries(now time.Time) {
 		}
 
 		newMatchersConfig := i.limits.ActiveSeriesCustomTrackersConfig(userID)
-		if newMatchersConfig.String() != userDB.activeSeries.CurrentConfig().String() {
+		newCostAttributionTracker := getCATrackerForUser(userID, i.costAttributionMgr)
+		if newMatchersConfig.String() != userDB.activeSeries.CurrentConfig().String() || newCostAttributionTracker != userDB.activeSeries.CurrentCostAttributionTracker() {
```

Comment:
How about we do something like:
```go
if userDB.activeSeries.ConfigDiffers(newMatchersConfig, newCostAttributionTracker)
```
?

Reply:
addressed in commit https://github.com/grafana/mimir/pull/9733/commits/f4ebc07871ef79a0ed7ee31c48032a9fd42fe757#diff-e1032332627c413a3010c66b54b22b6e9835cf152fa339e40cf0b11204f7241fR793